### ISSUE TITLE
Cypress 6.3.0 Fails to exit if connecting to Chrome falls back to TCP

### ISSUE THREAD

**Issue by rpong** [cypress-io/cypress#14701]
### Current behavior
Cypress 6.3.0 Running in Headless Mode, set to Chrome Browser does not EXIT after the tests are done, if the connection falls back to TCP, showing this in any part of the test.

```
Timed out waiting for the browser to connect. Retrying...
Warning: Cypress failed to connect to Chrome via stdio after 1 minute, 0 seconds. Falling back to TCP...
Connecting to Chrome via TCP was successful, continuing with tests
```

Additional Information:
- We have to run "killall -9 chrome" for the Cypress to exit, if the tests all passed, it does return a success status when doing this. So it is likely an issue with Cypress during termination.

- If there are retries in the test, which should have otherwise passed due to retry, the retry is not honored, and the test is reported as failed 

- This only happens in our AWS EC2 Linux Ubuntu 20.04, it DOES NOT happen in our Mac Mini environment, maybe because the Chrome TCP Fallback does not happen at all in our Mac-Mini Server.

### Desired behavior
Cypress 6.3.0 Running in Headless Mode, set to Chrome Browser SHOULD EXIT after the tests are done even if the connection falls back to TCP

### Test code to reproduce
This happens on any occasion that this notice appears:

```
Timed out waiting for the browser to connect. Retrying...
Warning: Cypress failed to connect to Chrome via stdio after 1 minute, 0 seconds. Falling back to TCP...
Connecting to Chrome via TCP was successful, continuing with tests
```

### Versions
6.3.0

**Comment by SimulatedGREG** [comment]
I can confirm the same issue with GitHub Actions, using `ubuntu-18.04`. Our tests are also running in headless mode against Chrome.

![image](https://user-images.githubusercontent.com/8327811/105871964-7918e900-5fbf-11eb-93a5-d16c36f9de79.png)

I can't link to the run because it is in a private repository. Hope this helps and gets resolved soon :slightly_smiling_face:. Our internal systems unfortunately hit our usage limit was hit very quickly because of this problem. Our normal 20 minutes runs turned into 4-6 hours runs :upside_down_face:.

**Comment by jalal246** [comment]
Have the same issue: https://github.com/jalal246/dflex/runs/1779823466?check_suite_focus=true with an error "**Error: Could not find Cypress test run results**"

**Comment by geremy22** [comment]
Having the same issue, test are executed twice in my windows 10 notebook with chrome 88:
![image](https://user-images.githubusercontent.com/8916503/106220985-72ef5c00-61bb-11eb-8c9d-7f3a13a2ad28.png)

And apparently running once on gitlab shared runner but with a huge timeout delay. [Here's job log](https://gitlab.com/while-true/typescript/injection-machine/-/jobs/994127486)

**Comment by emilong** [comment]
I can also confirm this happening on CircleCI. The change in https://github.com/cypress-io/cypress/pull/14348 looks like it might caused the trouble, but was reverted in https://github.com/cypress-io/cypress/pull/14819.

Sounds like reverting to v6.2.1 might help for now and the next release will also work?

**Comment by jbutko** [comment]
Same here, reverting from `6.3.0` to `6.2.0` fixed the problem.

**Comment by Ahmed-Abd-El-Ghany** [comment]
same over here getting this errror:

Timed out waiting for the browser to connect. Retrying...
Warning: Cypress failed to connect to Chrome via stdio after 1 minute, 0 seconds. Falling back to TCP...
Connecting to Chrome via TCP was successful, continuing with tests.

**Comment by giacaglia** [comment]
It seems that this was fixed on version 6.4.0: https://github.com/cypress-io/cypress/releases/tag/v6.4.0

**Comment by flotwig** [comment]
Closing since this is fixed in 6.4.0. Please open a new issue if you are still experiencing this.
